src/Internal.AspNetCore.CodeSign.Abstractions/SignRequestItem.cs

```diff
@@ -0,0 +1,171 @@
+            => new SignRequestItem(path, SignRequestItemType.Vsix)
+            {
+                Certificate =  certificateName,
+            };
+
+        /// <summary>
+        /// Determines if a file is signable based on the file extension
+        /// </summary>
+        /// <param name="filePath">The file extension (beginning with a dot). <seealso cref="System.IO.Path.GetExtension"/></param>
+        /// <returns>True if the path can be code-signed.</returns>
+        public static bool IsFileTypeSignable(string filePath)
+            => GetTypeFromFileExtension(System.IO.Path.GetExtension(filePath).TrimStart('.')) != SignRequestItemType.Unknown;
+
+        /// <summary>
+        /// Get an item type based on its file extension.
+        /// </summary>
+        /// <param name="fileExtension"></param>
+        /// <returns></returns>
+        public static SignRequestItemType GetTypeFromFileExtension(string fileExtension)
+        {
+            if (FileExtensionMapping.TryGetValue(fileExtension.TrimStart('.'), out var type))
+            {
+                return type;
+            }
+
+            return SignRequestItemType.Unknown;
+        }
+
+        /// <summary>
+        /// Initialize an instance of <see cref="SignRequestItem" />
+        /// </summary>
+        /// <param name="path"></param>
+        /// <param name="itemType"></param>
+        internal SignRequestItem(string path, SignRequestItemType itemType)
+        {
+            Path = path ?? throw new ArgumentNullException(nameof(path));
+            ItemType = itemType;
+        }
+
+        /// <summary>
+        /// The file path. Should be relative to the manifest or container.
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// The type of the item.
+        /// </summary>
+        public SignRequestItemType ItemType { get; }
+
+        /// <summary>
+        /// The name of the code signing certificate to use.
+        /// </summary>
+        public string Certificate { get; internal set; }
+
+        /// <summary>
+        /// The name of the strong name key to use (for .NET assemblies only).
+        /// </summary>
+        public string StrongName { get; internal set; }
+
+        /// <summary>
+        /// Excluded from code signing.
+        /// </summary>
+        public bool Excluded => ItemType == SignRequestItemType.Exclusion;
+
+        /// <summary>
+        /// True for item types which can contain children, such as .zip or .nupkg files.
+        /// </summary>
+        public bool CanContainChildren
+            => ItemType == SignRequestItemType.Nupkg
+               || ItemType == SignRequestItemType.Zip
+               || ItemType == SignRequestItemType.Vsix;
+
+        /// <summary>
+        /// Children of the sign request item.
+        /// </summary>
+        public IEnumerable<SignRequestItem> Children => _children;
```

Comment:
Should this be a `SignRequestCollection`? Why have `_children` be read-only but not `Children`?

Reply:
I wanted to hide the implementation. It shouldn't matter to the user how this collection is implemented.